app/models/grda_warehouse/warehouse_reports/hud_lot.rb

```diff
@@ -0,0 +1,326 @@
+          summary[key] ||= Set.new
+          summary[key] << type
+        end
+        summary.sort_by{|k, _|}.reverse
+      end
+    end
+
+    def details_by_range
+      @details_by_range ||= begin
+        details = []
+        (start_date, current_location) = locations_by_date&.first
+        locations_by_date.each_with_index do |(date, type), index|
+          next_date = date + 1.days
+          next_location = locations_by_date[next_date]
+          # we've reached the end
+          if next_location.blank?
+            details << {
+              start_date: start_date,
+              end_date: date,
+              type: current_location,
+            }
+          else
+            next if type == current_location
+
+            details << {
+              start_date: start_date,
+              end_date: date,
+              type: current_location,
+            }
+            start_date = next_date
+            current_location = next_location
+          end
+        end
+        details.reverse
+      end
+    end
+
+    private def dates
+      @dates ||= @filter.range.zip([]).to_h
+    end
+
+    private def ph_dates
+      @ph_dates ||= ph_services.distinct.pluck(:date).map do |d|
+        [
+          d,
+          ph_stay,
+        ]
+      end.to_h
+    end
+
+    private def th_dates
+      @th_dates ||= th_services.distinct.pluck(:date).map do |d|
+        [
+          d,
+          th_stay,
+        ]
+      end.to_h
+    end
+
+    private def literally_homeless_dates
+      @literally_homeless_dates ||= begin
+        lit_dates = literally_homeless_services.distinct.order(date: :asc).pluck(:date).map do |d|
+          [
+            d,
+            shelter_stay,
+          ]
+        end.to_h
+        extra_days = {}
+        # Fill in any gaps of < 7 days
+        lit_dates.to_a.each_with_index do |(date, _), i|
+          (next_date, _) = lit_dates.to_a[i + 1]
+          next if next_date.blank? || next_date > filter.end
+
+          if next_date < date + 7.days
+            (date..next_date).each do |d|
+              extra_days[d] = shelter_stay unless lit_dates.key?(d)
+            end
+          end
+        end
+
+        lit_dates.merge(extra_days).sort_by{|k,_| k}.to_h
+      end
+    end
+
+    # A break is added any time a PH/TH stay is started between two street stays
+    # and lasts more than 7 days
+    # 1. Occurs between two Documented street/shelter dates 7 or more days apart
+    # 2. AND does not include a Documented street/shelter date
+    # 3. AND includes 7 or more consecutive days in permanent or transitional housing.
+
+    private def breaks(un_processed_dates)
+      breaks = {}
+      a_dates = un_processed_dates.to_a
+      dates_present = a_dates.select{|_, v| v.present?}
+      dates_present.each_with_index do |(date, type), i|
+        next if shelter_stay?(type)
+        previous_i = i - 1
+        next unless shelter_stay?(dates_present[previous_i].last)
```

Comment:
In the event that the first entry is not a shelter stay, this would wrap around to the end of the array...